plugin/storage/grpc/shared/grpc_server.go

```diff
@@ -78,13 +78,24 @@ func (s *grpcServer) GetServices(ctx context.Context, r *storage_v1.GetServicesR
 }
 
 // GetOperations returns the operations of a given service
-func (s *grpcServer) GetOperations(ctx context.Context, r *storage_v1.GetOperationsRequest) (*storage_v1.GetOperationsResponse, error) {
-	operations, err := s.Impl.SpanReader().GetOperations(ctx, r.Service)
+func (s *grpcServer) GetOperations(ctx context.Context, r *storage_v1.GetOperationsRequest) (
+	*storage_v1.GetOperationsResponse, error) {
+	operations, err := s.Impl.SpanReader().GetOperations(ctx, &spanstore.OperationQueryParameters{
+		ServiceName: r.Service,
+		SpanKind:    r.SpanKind,
+	})
 	if err != nil {
 		return nil, err
 	}
+	grpcOperation := make([]*storage_v1.Operation, len(operations))
```

Comment:
is this backwards compatible if the plugin only returns operation names?

Reply:
It's backward compatible in that case, since the new return structure will just have empty spanKind.

But I do need to return the response in below structure:
```
storage_v1.GetOperationsResponse{
		Operations:   uniqueNames,
		OperationsV2: grpcOperation,
	}
```
So that old grpc client can still get unique operations names

Reply:
but I don't see you reading `operationNames` field. The old plugin will only return that field.

Reply:
actually, never mind, it's fine. The comment applies to grpc client, which is what calls potentially old plugin.

Reply:
In this PR, all plugins are updated to return []*spanReader.Operation struct{
	Name     string
	SpanKind string
}
For badger & ES, the spanKind will just be empty.

Are you saying it's possible for grpcService to be configured to use old version plugins?